### PR TITLE
WEBAPP-518: Fix api client flow types

### PR DESCRIPTION
**Pull request by Schedulaar**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

File: src/modules/endpoint/DatabaseConnector.js [hunk 1]
```diff
@@ -372,6 +374,8 @@ class DatabaseConnector {
         live: jsonObject.live,
         eventsEnabled: jsonObject.events_enabled,
         extrasEnabled: jsonObject.extras_enabled,
+        tunewsEnabled: false, // todo: NATIVE-549
```
Comment (klinzo) [inline src/modules/endpoint/DatabaseConnector.js:377]: Strange TODOs :thinking:

File: src/modules/endpoint/ResourceURLFinder.js [hunk 2]
```diff
@@ -5,9 +5,14 @@ import { Parser } from 'htmlparser2'
 import type { FetchMapType } from './sagas/fetchResourceCache'
 import { reduce } from 'lodash'
 import hashUrl from './hashUrl'
-import { ExtendedPageModel } from '@integreat-app/integreat-api-client'
 import Url from 'url-parse'
 
+interface InputEntryType {
```
Comment (klinzo) [inline src/modules/endpoint/ResourceURLFinder.js:10]: Why use `interface` here? What is the difference?

File: src/routes/events/components/Events.js [hunk 3]
```diff
@@ -121,8 +122,8 @@ class Events extends React.Component<PropsType> {
           <>
             <PageDetail identifier={t('date')} information={event.date.toFormattedString(language)}
                         theme={theme} language={language} />
-            {event.location.location && <PageDetail identifier={t('location')}
-                                                    information={event.location.location} theme={theme}
+            {location && <PageDetail identifier={t('location')}
+                                                    information={location} theme={theme}
```
Comment (klinzo) [inline src/routes/events/components/Events.js:126]: Indentation.

File: src/modules/endpoint/ResourceURLFinder.js (hunk 2 above)
Comment (Schedulaar) [reply src/modules/endpoint/ResourceURLFinder.js:10]: https://flow.org/en/docs/types/interfaces/
The benefit is, that I can test the method by only passing objects that have the necessary properties and so I don't have to create ExtendedPageModels with all the attributes that are not needed for (testing) this method.

File: src/modules/endpoint/DatabaseConnector.js (hunk 1 above)
Comment (Schedulaar) [reply src/modules/endpoint/DatabaseConnector.js:377]: Well, these attributes have to be properly and carefully filled into the database. I didn't want to do this in this task.

File: src/modules/endpoint/ResourceURLFinder.js (hunk 2 above)
Comment (klinzo) [reply src/modules/endpoint/ResourceURLFinder.js:10]: Should we mention this in the model builder issue then to change it?

File: src/modules/endpoint/DatabaseConnector.js (hunk 1 above)
Comment (klinzo) [reply src/modules/endpoint/DatabaseConnector.js:377]: Was more like `todo` vs `TODO` but oc that's fine :smile:

File: src/modules/endpoint/ResourceURLFinder.js (hunk 2 above)
Comment (Schedulaar) [reply src/modules/endpoint/ResourceURLFinder.js:10]: Not sure what you mean by that

File: src/modules/app/__tests__/CategoriesRouteStateView.spec.js [hunk 4]
```diff
@@ -14,7 +14,9 @@ describe('CategoriesRouteStateView', () => {
       parentPath: '',
       order: 0,
       availableLanguages: new Map(),
-      lastUpdate: moment('2017-11-18 19:30:00', moment.ISO_8601)
+      lastUpdate: moment('2017-11-18 19:30:00', moment.ISO_8601),
```
Comment (maxammann) [inline src/modules/app/__tests__/CategoriesRouteStateView.spec.js:17]: Use there the syntax: `2007-08-31T16:47+00:00`

File: src/routes/dashboard/containers/__tests__/DashboardContainer.spec.js [hunk 5]
```diff
@@ -82,6 +83,9 @@ describe('DashboardContainer', () => {
   }
 
   const rootCategory = categoriesMap.findCategoryByPath(`/${city.code}/${language.code}`)
+  if (!rootCategory) {
+    throw Error('The root category was not found. Sth\'s odd.')
```
Comment (maxammann) [inline src/routes/dashboard/containers/__tests__/DashboardContainer.spec.js:87]:
```suggestion
    throw Error('The root category was not found. Something is very odd.')
```

File: src/modules/endpoint/ResourceURLFinder.js (hunk 2 above)
Comment (klinzo) [reply src/modules/endpoint/ResourceURLFinder.js:10]: This (the use of interface) will be obsolete once we have model builders, right? If so, we could add a note to the model builder issue you created.

Comment (Schedulaar) [reply src/modules/endpoint/ResourceURLFinder.js:10]: Hmmm, no I wouldn't adjust this. Imo this is equally good or even better than using modelBuilders and ExtendedPageModel here, as we only need the 3 attributes defined in the interface for using the method...

**Pull request merged**